go/protocols/capture/pull_client_test.go

```diff
@@ -47,12 +47,14 @@ func TestPullClientLifecycle(t *testing.T) {
 	// |captured|, and reduces the driver checkpoint into |reducedCheckpoint|.
 	// It models the caller's expected behavior of producing captured documents
 	// into a collection upon notification.
-	var drain = func() {
+	var drain = func() string {
 		var combiner = rpc.Combiners()[0].(*pf.MockCombiner)
+		var n = len(combiner.Combined)
 		captured = append(captured, combiner.Combined...)
 		combiner.Combined = nil
 
 		require.NoError(t, reducedCheckpoint.Reduce(rpc.DriverCheckpoint()))
+		return fmt.Sprintf("%d => %s", n, string(reducedCheckpoint.DriverCheckpointJson))
```

Comment:
Is there something that guarantees a stable order of the keys in the serialized driver checkpoint? I seem to remember the keys being in lexicographical order, but I'm not 100%.

Reply:
Good question. I believe it's stable because novel keys are merged into the serialization at the end, which only happens to match the lexicographic order.